src/scripts/main.js

```diff
@@ -1,3 +1,189 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const tableBody = table.children[1];
+const tableHead = table.children[0];
+
+// sort table function
+
+const sortTable = (sortBy, order) => {
+  const index = [...tableHead.firstElementChild.children]
+    .findIndex(item => item.textContent === sortBy);
+  const rows = [...tableBody.children];
+
+  rows.sort((a, b) => {
+    let valueA = a.children[index].textContent;
+    let valueB = b.children[index].textContent;
+
+    valueB = !order ? [valueA, valueA = valueB][0] : valueB;
+
+    if (+valueA) {
+      return valueA - valueB;
+    }
+
+    if (+valueA.slice(1).split(',').join('')) {
+      return +valueA.slice(1).split(',').join('')
+      - +valueB.slice(1).split(',').join('');
+    }
+
+    return valueA > valueB ? 1 : valueA < valueB ? -1 : 0;
+  });
+
+  rows.forEach(item => {
+    tableBody.append(item);
+  });
+};
+
+// sort table by click on header
+
+let onceClicked;
+let clickCount = 0;
+
+tableHead.firstElementChild.onclick = (e) => {
+  onceClicked === e.target ? clickCount++ : clickCount = 1;
+
+  sortTable(e.target.innerText, clickCount % 2);
+  onceClicked = e.target;
+};
+
+// selected row class active
+
+tableBody.onclick = (e) => {
+  [...tableBody.children].forEach(item => {
+    item.classList.remove('active');
+  });
+  e.target.parentElement.classList.add('active');
+};
+
+// new Form
+
+const newEmployeeForm = document.createElement('form');
+
+newEmployeeForm.classList.add('new-employee-form');
+
+newEmployeeForm.insertAdjacentHTML('afterbegin', `
+<label>Name: <input name="name" type="text" data-qa="name" required></label>
+<label>Position:
+  <input name="position" type="text" data-qa="position" required>
+</label>
+<label>Office:
+  <select name="office" data-qa="office" required>
+    <option>Tokyo</option>
+    <option>Singapore</option>
+    <option>London</option>
+    <option>New York</option>
+    <option>Edinburgh</option>
+    <option>San Francisco</option>
+  </select>
+</label>
+<label>Age: <input name="age" type="number" data-qa="age" required></label>
+<label>Salary:
+    <input name="salary" type="number"  data-qa="salary" required>
+</label>
+<button type="submit">Save to table</button>
+`);
+document.body.append(newEmployeeForm);
+
+// validate function
+
+const validateForm = function(elements) {
+  elements.pop();
+
+  switch (true) {
```

Comment:
I would recommend to refactor this validation function:
1) it's better not use switch statement like this. You can use just early return
Example: 
```
  const isEmpty = rest.some(item => !item.value.length);
  if (isEmpty) {
    pushNotification('Error', 'All fields are required', 'error');
    return false;
  }
```
2) Use a named constant for the minimum name length, minimum and maximum age (outside the function)
```
  const MIN_NAME_LENGTH = 4;
  const MIN_AGE = 18;
  const MAX_AGE = 90;
  ```
  3) just create a variables for conditions
  Example:
  ```
  const isNameTooShort = rest[0].value.length < MIN_NAME_LENGTH;
  ```